### PR TITLE
Changing ITestPlatform interface and adding Protocol config in request Data

### PR DESCRIPTION
**Pull request by khilangoel**
Added request data in ITestPlatform interface. 
Added protocol config in request data itself.

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/RequestData.cs [hunk 1]
```diff
@@ -5,28 +5,42 @@ namespace Microsoft.VisualStudio.TestPlatform.Common
...
-        /// <inheritdoc/>
-        public IMetricsCollection MetricsCollection => this.metricsCollection;
+        /// <inheritdoc />
+        /// <summary>
+        /// Gets or sets the protocol config.
+        /// </summary>
+        public ProtocolConfig ProtocolConfig
+        {
+            get => this.protocolConfig;
+            set => this.protocolConfig = value;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/RequestData.cs]: Add null check here as well

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs [hunk 2]
```diff
@@ -14,13 +14,13 @@ namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Discovery
...
     using Microsoft.VisualStudio.TestPlatform.Common.Filtering;
     using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
-    using Microsoft.VisualStudio.TestPlatform.Common.Interfaces.Engine;
     using Microsoft.VisualStudio.TestPlatform.Common.Logging;
     using Microsoft.VisualStudio.TestPlatform.Common.Telemetry;
     using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Tracing;
     using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Tracing.Interfaces;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs]: Why do we need this?

**Review by singhsarab — APPROVED** [review]

File: src/Microsoft.TestPlatform.Common/RequestData.cs (hunk 1 above)
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.Common/RequestData.cs]: done

File: src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs [hunk 3]
```diff
@@ -52,19 +51,18 @@ protected TestEngine(TestRuntimeProviderManager testHostProviderManager, IProces
         /// Fetches the DiscoveryManager for this engine. This manager would provide all functionality required for discovery.
         /// </summary>
         /// <param name="requestData">
-        /// The request data for providing common discovery services and data.
+        ///     The request data for providing common discovery services and data.
         /// </param>
         /// <param name="testHostManager">
-        /// Test host manager
+        ///     Test host manager
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs]: Nit : remove extra white spces.

File: src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs [hunk 4]
```diff
@@ -52,19 +51,18 @@ protected TestEngine(TestRuntimeProviderManager testHostProviderManager, IProces
         /// Fetches the DiscoveryManager for this engine. This manager would provide all functionality required for discovery.
         /// </summary>
         /// <param name="requestData">
-        /// The request data for providing common discovery services and data.
+        ///     The request data for providing common discovery services and data.
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs]: nit : Remove extra white spaces.

**Review by harshjain2 — APPROVED** [review]

**Comment by khilan2goel** [comment]
@dotnet-bot test this please

**Comment by khilan2goel** [comment]
@dotnet-bot test this please

**Pull request merged**